ilc/client/GuardManager.js

```diff
@@ -0,0 +1,112 @@
+const navigateToUrl = require('single-spa').navigateToUrl;
+const unlocalizeUrl = require('../common/i18n/unlocalizeUrl');
+
+const actionTypes = require('../common/guard/actionTypes');
+
+const getIlcRouter = require('./getIlcRouter');
+const getIlcConfig = require('./ilcConfig');
+
+class GuardManager {
+    #registryConf;
+    #router;
+    #unlocalizeUrl;
+    #navigateToUrl;
+    #transitionHooksPlugin;
+
+    constructor(
+        registryConf,
+        router,
+        unlocalizeUrl,
+        navigateToUrl,
+        transitionHooksPlugin,
+    ) {
+        this.#registryConf = registryConf;
+        this.#router = router;
+        this.#unlocalizeUrl = unlocalizeUrl;
+        this.#navigateToUrl = navigateToUrl;
+        this.#transitionHooksPlugin = transitionHooksPlugin;
+    }
+
+    shouldNavigateTo(url) {
+        const pathname = new URL(url, 'http://hack').pathname;
+        const route = this.#router.match(this.#unlocalizeUrl(this.#registryConf.settings.i18n, pathname));
```

Comment:
This looks like a bad design decision... We use common/Router here directly & totally ignore existence of the ClientRouter... So we expect `match()` fm here return same result as ClientRouter will get - but this may not be true as we already have some "preparation" code (unlocalizeUrl) which is copy pasted here from ClientRouter.